datafusion/optimizer/src/simplify_expressions/unwrap_cast.rs

```diff
@@ -735,10 +738,12 @@ mod tests {
     }
 
     fn optimize_test(expr: Expr, schema: &DFSchemaRef) -> Expr {
-        let mut expr_rewriter = UnwrapCastExprRewriter {
-            schema: Arc::clone(schema),
-        };
-        expr.rewrite(&mut expr_rewriter).data().unwrap()
+        let props = ExecutionProps::new();
```

Comment:
The original tests looked like the following

```rust
    fn optimize_test(expr: Expr, schema: &DFSchemaRef) -> Expr {
        let mut expr_rewriter = UnwrapCastExprRewriter {
            schema: Arc::clone(schema),
        };
        expr.rewrite(&mut expr_rewriter).data().unwrap()
    }
```

I changed `optimize_test` to use `ExprSimplifier` instead and most of the test just pass unchanged